packages/widget/src/hooks/useToAddressRequirements.ts

```diff
@@ -3,18 +3,29 @@ import { useWidgetConfig } from '../providers';
 import { RequiredUI } from '../types';
 import { useFieldValues } from '../stores';
 
-export const useRequiredToAddress = () => {
+export const useToAddressRequirements = () => {
   const { requiredUI } = useWidgetConfig();
-  const [fromChainId, toChainId] = useFieldValues('fromChain', 'toChain');
+  const [fromChainId, toChainId, fromToken, toToken] = useFieldValues(
+    'fromChain',
+    'toChain',
+    'fromToken',
+    'toToken',
+  );
 
   const { chain: fromChain } = useChain(fromChainId);
   const { chain: toChain } = useChain(toChainId);
 
+  const requiredChainType =
+    !!fromToken && !!toToken && !!fromChain && !!toChain && toChain.chainType;
```

Comment:
Can we potentially have chains without chainType? If so I guess we need to check for fromChain.chainType as well